src/Resources/ResourceManager/Implementation/Policy/SetAzurePolicyExemption.cs

```diff
@@ -0,0 +1,186 @@
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PolicyExemption", DefaultParameterSetName = PolicyCmdletBase.NameParameterSet), OutputType(typeof(PsPolicyExemption))]
+    public class SetAzurePolicyExemptionCmdlet : PolicyCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the policy exemption name parameter.
+        /// </summary>
+        [Parameter(ParameterSetName = PolicyCmdletBase.NameParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionNameHelp)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the policy exemption scope parameter
+        /// </summary>
+        [Parameter(ParameterSetName = PolicyCmdletBase.NameParameterSet, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionScopeHelp)]
+        [ValidateNotNullOrEmpty]
+        public string Scope { get; set; }
+
+        /// <summary>
+        /// Gets or sets the policy assignment id parameter
+        /// </summary>
+        [Alias("ResourceId")]
+        [Parameter(ParameterSetName = PolicyCmdletBase.IdParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionIdHelp)]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the policy exemption display name parameter
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionDisplayNameHelp)]
+        [ValidateNotNullOrEmpty]
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the policy exemption description parameter
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionDescriptionHelp)]
+        [ValidateNotNullOrEmpty]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// Gets or sets the policy exemption category
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionCategoryHelp)]
+        [ValidateSet(PolicyExemptionCategory.Waiver, PolicyExemptionCategory.Mitigated)]
+        public string ExemptionCategory { get; set; }
+
+        /// <summary>
+        /// Gets or sets the policy definition reference ID list when the associated policy assignment is for a policy set (initiative).
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = PolicyHelpStrings.SetPolicyExemptionPolicyDefinitionReferenceIdsHelp)]
+        [ValidateNotNullOrEmpty]
+        public string[] PolicyDefinitionReferenceIds { get; set; }
```

Comment:
1. Does not support `ShouldProcess` but the cmdlet verb Set indicates that it should.
2. Parameter `PolicyDefinitionReferenceIds` of cmdlet `Set-AzPolicyExemption` does not follow the enforced naming convention of using a singular noun for a parameter name.

Reply:
1. Added ShouldProcess support
2. PolicyDefinitionReferenceIds is a policy term. It cannot be changed. Here is a document reference. https://docs.microsoft.com/en-us/azure/governance/policy/concepts/initiative-definition-structure#policy-definition-properties